Fix cache test to not rely on current context

<!-- 
  Issues without logs and details are more complicated to fix.
  Please help us by filling the template below!
-->
Skaffold Build Cache tests ```github.com/GoogleContainerTools/skaffold/pkg/skaffold/build/cache``` fails on master at or above **b0f4b0315aa947dd26e32e6bfdce294e18b0eecf** if kubeconfig context is set to minikube.

### Expected behavior
`go test  ./...` should pass irrespective of what the current context is set to in user's development environment

### Actual behavior
```
tejaldesai@tejaldesai~/go/src/github.com/GoogleContainerTools/skaffold: (master)$ kubectl config use-context minikube
Switched to context "minikube".
tejaldesai@tejaldesai~/go/src/github.com/GoogleContainerTools/skaffold: (master)$ kubectl config current-context
minikube

tejaldesai@tejaldesai~/go/src/github.com/GoogleContainerTools/skaffold: (master)$ go clean --testcache
tejaldesai@tejaldesai~/go/src/github.com/GoogleContainerTools/skaffold: (master)$ go test -v github.com/GoogleContainerTools/skaffold/pkg/skaffold/build/cache -run ^Test_NewCache
=== RUN   Test_NewCache
=== RUN   Test_NewCache/get_a_valid_cache_from_file
time="2019-03-12T15:05:51-07:00" level=info msg="no config entry found for kube-context minikube"
=== RUN   Test_NewCache/needs_push
time="2019-03-12T15:05:51-07:00" level=info msg="no config entry found for kube-context minikube"
=== RUN   Test_NewCache/valid_cache_file_exists,_but_useCache_is_false
=== RUN   Test_NewCache/corrupted_cache_file
time="2019-03-12T15:05:51-07:00" level=warning msg="Error retrieving artifact cache, not using skaffold cache: yaml: unmarshal errors:\n  line 1: cannot unmarshal !!str `corrupt...` into cache.ArtifactCache"
--- FAIL: Test_NewCache (0.00s)
    --- FAIL: Test_NewCache/get_a_valid_cache_from_file (0.00s)
        cache_test.go:148: Expected result different from actual result. Expected: &{map[hash:{digest id}] 0xc00028c0a0 <nil> [{0 0 image map[]  [] [] 0 0 0}] /tmp/827158023 true false false}, Actual: &{map[hash:{digest id}] 0xc00028c140 <nil> [{0 0 image map[]  [] [] 0 0 0}] /tmp/827158023 true false true}
    --- FAIL: Test_NewCache/needs_push (0.00s)
        cache_test.go:148: Expected result different from actual result. Expected: &{map[hash:{digest id}] 0xc0000caf50 <nil> [] /tmp/649495994 true true false}, Actual: &{map[hash:{digest id}] 0xc0000cafa0 <nil> [] /tmp/649495994 true true true}
    --- PASS: Test_NewCache/valid_cache_file_exists,_but_useCache_is_false (0.00s)
    --- PASS: Test_NewCache/corrupted_cache_file (0.00s)
FAIL
FAIL	github.com/GoogleContainerTools/skaffold/pkg/skaffold/build/cache	0.117s
tejaldesai@tejaldesai~/go/src/github.com/GoogleContainerTools/skaffold: (master)$ 
``` 

### Information

- Skaffold version: skaffold @ b0f4b0315aa947dd26e32e6bfdce294e18b0eecf
- Operating system: ...
- Contents of skaffold.yaml:

```yaml
<paste your skaffold.yaml here>
```

### Steps to reproduce the behavior

1.  set kubecontext to minikube.
2.  run `go clean --testcache`
3. run `go test -v github.com/GoogleContainerTools/skaffold/pkg/skaffold/build/cache -run ^Test_NewCache`

--- a/pkg/skaffold/build/cache/cache_test.go
+++ b/pkg/skaffold/build/cache/cache_test.go
@@ -30,6 +30,7 @@ import (
 	"github.com/GoogleContainerTools/skaffold/testutil"
 	"github.com/docker/docker/api/types"
 	yaml "gopkg.in/yaml.v2"
+	"k8s.io/client-go/tools/clientcmd/api"
 )
 
 var (
@@ -122,6 +123,8 @@ func Test_NewCache(t *testing.T) {
 
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
+			restore := testutil.SetupFakeKubernetesContext(t, api.Config{CurrentContext: "cluster1"})
+			defer restore()
 
 			cacheFile := createTempCacheFile(t, test.cacheFileContents)
 			if test.updateCacheFile {

--- a/pkg/skaffold/runner/dev_test.go
+++ b/pkg/skaffold/runner/dev_test.go
@@ -28,6 +28,7 @@ import (
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/sync"
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/watch"
 	"github.com/GoogleContainerTools/skaffold/testutil"
+	"k8s.io/client-go/tools/clientcmd/api"
 )
 
 type NoopWatcher struct{}
@@ -92,6 +93,9 @@ func discardOutput() *config.Output {
 }
 
 func TestDevFailFirstCycle(t *testing.T) {
+	restore := testutil.SetupFakeKubernetesContext(t, api.Config{CurrentContext: "cluster1"})
+	defer restore()
+
 	var tests = []struct {
 		description     string
 		testBench       *TestBench
@@ -148,6 +152,9 @@ func TestDevFailFirstCycle(t *testing.T) {
 }
 
 func TestDev(t *testing.T) {
+	restore := testutil.SetupFakeKubernetesContext(t, api.Config{CurrentContext: "cluster1"})
+	defer restore()
+
 	var tests = []struct {
 		description     string
 		testBench       *TestBench
@@ -280,6 +287,9 @@ func TestDev(t *testing.T) {
 }
 
 func TestDevSync(t *testing.T) {
+	restore := testutil.SetupFakeKubernetesContext(t, api.Config{CurrentContext: "cluster1"})
+	defer restore()
+
 	var tests = []struct {
 		description     string
 		testBench       *TestBench

--- a/pkg/skaffold/runner/runner_test.go
+++ b/pkg/skaffold/runner/runner_test.go
@@ -290,6 +290,9 @@ func TestNewForConfig(t *testing.T) {
 }
 
 func TestRun(t *testing.T) {
+	restore := testutil.SetupFakeKubernetesContext(t, api.Config{CurrentContext: "cluster1"})
+	defer restore()
+
 	var tests = []struct {
 		description     string
 		testBench       *TestBench

